src/main/java/ua/com/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package ua.com.cinema.security;
+
+import ua.com.cinema.exception.AuthenticationException;
+import ua.com.cinema.lib.Inject;
+import ua.com.cinema.lib.Service;
+import ua.com.cinema.model.User;
+import ua.com.cinema.service.impl.UserService;
+import ua.com.cinema.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User user = userService.findByEmail(email).orElseThrow(()
+                -> new AuthenticationException("User with this email was not found"));
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (hashedPassword.equals(user.getPassword())) {
+            return user;
+        }
+        throw new AuthenticationException("Wrong password");
```

Comment:
let's refactor this method so that we could throw the exception only once with the message `The login or password were wrong` if any of the conditions is not met

we don't want the person to know the exact reason of failure in this case - what if someone wants to login as another person?